File: src/main/java/kr/co/lunatalk/global/util/CookieUtil.java
package kr.co.lunatalk.global.util;

import jakarta.servlet.http.Cookie;
import jakarta.servlet.http.HttpServletResponse;
import kr.co.lunatalk.infra.config.jwt.JwtProperties;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class CookieUtil {

	private final JwtProperties jwtProperties;
	private final SpringEnvironmentUtil springEnvironmentUtil;

	public void setAccessTokenCookie(HttpServletResponse response, String accessToken) {
		Cookie cookie = new Cookie("accessToken", accessToken);
		cookie.setHttpOnly(true);
		cookie.setSecure(!springEnvironmentUtil.isLocalProfile()); // 로컬에서는 false, dev/prod에서는 true
		cookie.setPath("/");
		cookie.setMaxAge((int) (jwtProperties.getAccessTokenExpirationTime() / 1000)); // 초 단위로 변환
		response.addCookie(cookie);
	}

	public void setRefreshTokenCookie(HttpServletResponse response, String refreshToken) {
		Cookie cookie = new Cookie("refreshToken", refreshToken);
		cookie.setHttpOnly(true);
		cookie.setSecure(!springEnvironmentUtil.isLocalProfile()); // 로컬에서는 false, dev/prod에서는 true
		cookie.setPath("/");
		cookie.setMaxAge((int) (jwtProperties.getRefreshTokenExpirationTime() / 1000)); // 초 단위로 변환
		response.addCookie(cookie);
	}

	public void deleteAccessTokenCookie(HttpServletResponse response) {
		Cookie cookie = new Cookie("accessToken", null);
		cookie.setHttpOnly(true);
		cookie.setSecure(!springEnvironmentUtil.isLocalProfile());
		cookie.setPath("/");
		cookie.setMaxAge(0);
		response.addCookie(cookie);
	}

	public void deleteRefreshTokenCookie(HttpServletResponse response) {
		Cookie cookie = new Cookie("refreshToken", null);
		cookie.setHttpOnly(true);
		cookie.setSecure(!springEnvironmentUtil.isLocalProfile());
		cookie.setPath("/");
		cookie.setMaxAge(0);
		response.addCookie(cookie);
	}
}

File: src/main/kotlin/kr/co/lunatalk/global/util/CookieUtil.kt
package kr.co.lunatalk.global.util

import jakarta.servlet.http.Cookie
import jakarta.servlet.http.HttpServletResponse
import kr.co.lunatalk.infra.config.jwt.JwtProperties
import org.springframework.stereotype.Component

@Component
class CookieUtil(
    private val jwtProperties: JwtProperties,
    private val springEnvironmentUtil: SpringEnvironmentUtil
) {

    fun setAccessTokenCookie(response: HttpServletResponse, accessToken: String) {
        val cookie = Cookie("accessToken", accessToken).apply {
            isHttpOnly = true
            secure = !springEnvironmentUtil.isLocalProfile() // 로컬에서는 false, dev/prod에서는 true
            path = "/"
            maxAge = (jwtProperties.accessTokenExpirationTimeMillis() / 1000).toInt()
        }
        response.addCookie(cookie)
    }

    fun setRefreshTokenCookie(response: HttpServletResponse, refreshToken: String) {
        val cookie = Cookie("refreshToken", refreshToken).apply {
            isHttpOnly = true
            secure = !springEnvironmentUtil.isLocalProfile() // 로컬에서는 false, dev/prod에서는 true
            path = "/"
            maxAge = (jwtProperties.refreshTokenExpirationTimeMillis() / 1000).toInt()
        }
        response.addCookie(cookie)
    }

    fun deleteAccessTokenCookie(response: HttpServletResponse) {
        val cookie = Cookie("accessToken", null).apply {
            isHttpOnly = true
            secure = !springEnvironmentUtil.isLocalProfile()
            path = "/"
            maxAge = 0
        }
        response.addCookie(cookie)
    }

    fun deleteRefreshTokenCookie(response: HttpServletResponse) {
        val cookie = Cookie("refreshToken", null).apply {
            isHttpOnly = true
            secure = !springEnvironmentUtil.isLocalProfile()
            path = "/"
            maxAge = 0
        }
        response.addCookie(cookie)
    }
}
